From a42f835c4fa683f33529bc9e1a2acad6835ab3bc Mon Sep 17 00:00:00 2001
From: raeno <just.raeno@gmail.com>
Date: Wed, 3 Dec 2014 21:50:58 +0300
Subject: [PATCH] Parameters set with options are same as specified via query
 string. Anyway we normalize them so it doesn't matter what we have, string or
 array with one string element.

---
 test/test_typhoeus_request_proxy.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/test_typhoeus_request_proxy.rb b/test/test_typhoeus_request_proxy.rb
index 9c11857..81e0d20 100644
--- a/test/test_typhoeus_request_proxy.rb
+++ b/test/test_typhoeus_request_proxy.rb
@@ -34,7 +34,7 @@ def test_that_proxy_simple_post_request_works_with_form_data
       :headers => {'Content-Type' => 'application/x-www-form-urlencoded'})
     request_proxy = OAuth::RequestProxy.proxy(request, {:uri => 'http://example.com/test'})
 
-    expected_parameters = {'key' => 'value'}
+    expected_parameters = {'key' => ['value']}
     assert_equal expected_parameters, request_proxy.parameters_for_signature
     assert_equal 'http://example.com/test', request_proxy.normalized_uri
     assert_equal 'POST', request_proxy.method
@@ -68,7 +68,7 @@ def test_that_proxy_post_request_works_with_mixed_parameter_sources
       :headers => {'Content-Type' => 'application/x-www-form-urlencoded'})
     request_proxy = OAuth::RequestProxy.proxy(request, {:uri => 'http://example.com/test?key=value', :parameters => {'key3' => 'value3'}})
 
-    expected_parameters = {'key' => ['value'], 'key2' => 'value2', 'key3' => 'value3'}
+    expected_parameters = {'key' => ['value'], 'key2' => ['value2'], 'key3' => 'value3'}
     assert_equal expected_parameters, request_proxy.parameters_for_signature
     assert_equal 'http://example.com/test', request_proxy.normalized_uri
     assert_equal 'POST', request_proxy.method
